cdap-watchdog/src/main/java/co/cask/cdap/metrics/store/timeseries/FactCodec.java

```diff
@@ -118,16 +125,39 @@ public FactCodec(EntityTable entityTable, int resolution, int rollTimebaseInterv
     return rowKey;
   }
 
+  public byte[] getNextRowKey(List<TagValue> previousTags, byte[] rowKey) {
```

Comment:
I'd change signature to:

``` java
public byte[] getNextRowKey(byte[] rowKey, int indexOfTagValueToChange)
```

May be you have better idea. But I guess what method does is, given the row key, suggests next row key that has different tagValue at given position. Sounds right or?